File: web/src/plugin/app/api/pointsConfig.js
import service from '@/utils/request'

export const createPointsConfig = (data) => service({ url: '/pcfg/createPointsConfig', method: 'post', data })
export const deletePointsConfig = (params) => service({ url: '/pcfg/deletePointsConfig', method: 'delete', params })
export const deletePointsConfigByIds = (params) => service({ url: '/pcfg/deletePointsConfigByIds', method: 'delete', params })
export const updatePointsConfig = (data) => service({ url: '/pcfg/updatePointsConfig', method: 'put', data })
export const findPointsConfig = (params) => service({ url: '/pcfg/findPointsConfig', method: 'get', params })
export const getPointsConfigList = (params) => service({ url: '/pcfg/getPointsConfigList', method: 'get', params })
export const getPointsConfigPublic = (params) => service({ url: '/pcfg/getPointsConfigPublic', method: 'get', params })

File: server/plugin/app/router/pointsConfig.go
package router

import (
    "github.com/flipped-aurora/gin-vue-admin/server/middleware"
    "github.com/gin-gonic/gin"
)

var PointsCfg = new(PCFG)

type PCFG struct{}

func (r *PCFG) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
    {
        group := private.Group("pcfg").Use(middleware.OperationRecord())
        group.POST("createPointsConfig", apiPointsCfg.CreatePointsConfig)
        group.DELETE("deletePointsConfig", apiPointsCfg.DeletePointsConfig)
        group.DELETE("deletePointsConfigByIds", apiPointsCfg.DeletePointsConfigByIds)
        group.PUT("updatePointsConfig", apiPointsCfg.UpdatePointsConfig)
    }
    {
        group := private.Group("pcfg")
        group.GET("findPointsConfig", apiPointsCfg.FindPointsConfig)
        group.GET("getPointsConfigList", apiPointsCfg.GetPointsConfigList)
    }
    {
        group := public.Group("pcfg")
        group.GET("getPointsConfigPublic", apiPointsCfg.GetPointsConfigPublic)
    }
}

